main.cpp

```diff
@@ -411,6 +523,37 @@ void Camera::Lens::getFocus(double distanceToObject)
     }
 }
 
+void Camera::adjustWhiteBalanceMode(int target)
+{
+    while (currentWhiteBalanceMode != target) //adjust in step of 5 unit. Assuming the focal length is product of 5
+    {
+        if (currentWhiteBalanceMode >= numberOfWhiteBalanceMode)
+        {
+            currentWhiteBalanceMode = numberOfWhiteBalanceMode;
+            return;
+        }
+        
+        if (currentWhiteBalanceMode < 1)
+        {
+            currentWhiteBalanceMode = 1;
+            return;
+        }
+
+        if (currentWhiteBalanceMode < target)
+        {   
+            std::cout << "Current white balance mode: " << currentWhiteBalanceMode << std::endl;
+            currentWhiteBalanceMode++;
+        }
+        else if (currentWhiteBalanceMode > target)
+        {
+            std::cout << "Current white balance mode: " << currentWhiteBalanceMode << std::endl;
+            currentWhiteBalanceMode--;
+        }
```

Comment:
Everywhere that you have `i++`, `i--`, `i -= 1;` or `i += 1;`, you should use `pre-increment/decrement`. 
You are currently doing `post-increment/decrement`.  
you can think of post- and pre- increment like this:
```
int preIncrement(int& i)
{
    i += 1;
    return i; //both the returned value AND i have changed
}
​
int postIncrement(int& i)
{
    int oldVal = i;
    i += 1;
    return oldVal; //i has changed but the value being returned does not reflect this change.
}
```
more information can be found here explaining why `pre-increment/decrement` is preferred over `post-increment/decrement`: https://en.cppreference.com/w/cpp/language/operator_incdec
- [ ] revise

Reply:
Thank you for this review and the info!